packages/core/src/shared/errors.ts

```diff
@@ -984,6 +984,65 @@ export class AwsClientResponseError extends Error {
     }
 }
 
+/**
+ * Represents a generalized error that happened during a disk cache operation.
+ *
+ * For example, when SSO refreshes a token a disk cache error can occur when it
+ * attempts to read/write the disk cache. These errors can be recoverable and do not
+ * imply that the SSO session is stale. So by wrapping those errors in an instance of
+ * this class it will help to distinguish them.
+ */
+export class DiskCacheError extends Error {
```

Comment:
what is specific to "cache" here? why is it not `DiskError`? (or `FilesystemError`) ?

Reply:
The intent of wrapping something that is ultimately a file system error in disk cache error is to give context to up upstream caller that this file system error is specific to disk cache.

The reason is that somewhere down the call stack may be an unrelated file system error (eg failed to write to the log file). At the top level when it receives this thrown error, the error does not clearly show where it originates from (log file vs disk cache).

So by wrapping file system errors in a a Disk Cache error the intent was for the receiver to know where that file system error came from. The alternative would be for them to follow the stack trace, but iirc doesn't work in production and takes more effort.

This allows the upstream to appropriately react to the "right" kind of file system error.

---

But I do agree it makes more sense to add these specific file system errors in to isFileSystemError and then DiskCacheError uses that under the hood.